tools/azure-devtools/src/azure_devtools/scenario_tests/recording_processors.py

```diff
@@ -224,6 +216,7 @@ def _process_body_as_dict(self, body):
         return new_body
 
     def process_response(self, response):
+        response = _decompress_response_body(response)
```

Comment:
What do you think of doing this in `ReplayableTest._process_response_recording`? That is to say, decompressing the body before running any processors? No processor would need a change then, because they would all get the decompressed body they expect.